lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/FeatureService.java

```diff
@@ -529,4 +534,149 @@ public List<Feature> readFeaturesFromFeatureListFile(List<Feature> installedFeat
         }
         return installedFeatures;
     }
+
+    /**
+     * get all platforms for all features from feature json
+     * @param libertyVersion liberty version
+     * @param libertyRuntime runtime
+     * @param requestDelay request delay
+     * @param documentURI document uri link
+     * @return set of unique platforms
+     */
+    public Set<String> getAllPlatforms(String libertyVersion, String libertyRuntime, int requestDelay, String documentURI) {
+        List<Feature> features = this.getFeatures(libertyVersion, libertyRuntime, requestDelay, documentURI);
+        return features.stream()
+                .map(Feature::getWlpInformation)
+                .filter(Objects::nonNull)
+                .map(WlpInformation::getPlatforms)
```

Comment:
Will this work ok if it is an older runtime and there are no platforms in the features.json? Is this just collecting / combining the empty Maps?

Reply:
It will return empty map